docs/modules/ROOT/pages/account-abstraction.adoc

```diff
@@ -0,0 +1,194 @@
+
+To build your own factory, see xref:account-abstraction.adoc#account_factory[Account Factory]
+
+=== Paymaster Contract
+
+A Paymaster is an optional entity that can sponsor gas fees for Accounts, or allow them to pay for those fees in ERC-20 instead of native currency. This abstracts gas away of the user experience in the same way that computational costs of cloud servers are abstracted away from end-users.
+
+To build your own paymaster, see xref:account-abstraction.adoc#paymaster[Paymaster].
+
+== Smart Accounts
+
+OpenZeppelin provides an abstract implementation of an xref:api:account.adoc#AccountBase[`AccountBase`] that implements both interfaces and accepts native currency and common tokens (i.e. xref:erc20.adoc[ERC-20], xref:erc721.adoc[ERC-721] and xref:erc1155.adoc[ERC-1155]), as well as validating any arbitrary `UserOperation`.
+
+Since developers should implement xref:api:account.adoc#AccountBase-validateUserOp-struct-PackedUserOperation-bytes32-uint256-[`_validateUserOp`] to use the account, the library provides 2 mechanisms to validate user operations. The user should select one of these to build an account.
+
+- **Modules**: An account that is controlled by validator and executor modules.
+- **Signers**: An account that is controlled by a private key.
+
+NOTE: Examples provided are simple version of accounts. When using with ERC-4337 infrastructure, a few adjustments are required to make the contract initializable and deployable by a factory. See the xref:account-abstraction.adoc#account_factory[account factory section] for more.
+
+An smart account can be created by deriving from the xref:api:account.adoc#AccountBase[`AccountBase`]:
+
+```solidity
+include::api:example$account/MyAccountECDSAPlainSigner.sol[]
+```
+
+NOTE: Using xref:api:utils.adoc#MessageHashUtils-toEthSignedMessageHash-bytes32-[`MessageHashUtils.toEthSignedMessageHash`] wraps the message with an `\x19Ethereum Signed Message:\n` prefix as most wallets do. However, this is not the most recommended approach since it requires the private key owner to sign an opaque message without contextual information. See xref:account-abstraction.adoc#readable_typed_signatures[readable typed signatures].
+
+==== ERC-7562 Validation Rules
+
+To process a bundle of `UserOperations`, bundlers call xref:api:account.adoc#AccountBase-validateUserOp-struct-PackedUserOperation-bytes32-uint256-[`validateUserOp`] on each operation sender to check whether the operation can be executed. However, the bundler has no guarantee that the state of the blockchain will remain the same after the validation phase. To overcome this problematic, https://eips.ethereum.org/EIPS/eip-7562[ERC-7562] proposes a set of limitations to EVM code so that bundlers (or node operators) are protected from unexpected state changes.
```

Comment:
```suggestion
To process a bundle of `UserOperations`, bundlers call xref:api:account.adoc#AccountBase-validateUserOp-struct-PackedUserOperation-bytes32-uint256-[`validateUserOp`] on each operation sender to check whether the operation can be executed. However, the bundler has no guarantee that the state of the blockchain will remain the same after the validation phase. To overcome this problem, https://eips.ethereum.org/EIPS/eip-7562[ERC-7562] proposes a set of limitations to EVM code so that bundlers (or node operators) are protected from unexpected state changes.
```